src/main/java/appeng/util/helpers/ItemComparisonHelper.java

```diff
@@ -0,0 +1,443 @@
+	 * If not it will also compare both damage values.
+	 * 
+	 * Ignores NBT.
+	 * 
+	 * @return true, if both are equal.
+	 */
+	public boolean isEqualItemType( final ItemStack that, final ItemStack other )
+	{
+		if( that != null && other != null && that.getItem() == other.getItem() )
+		{
+			if( that.isItemStackDamageable() )
+			{
+				return true;
+			}
+			return that.getItemDamage() == other.getItemDamage();
+		}
+		return false;
+	}
+
+	/**
+	 * A wrapper around {@link ItemStack#isItemEqual(ItemStack)}.
+	 * 
+	 * The benefit is to compare two null item stacks, without any additional null checks.
+	 * 
+	 * Ignores NBT.
+	 * 
+	 * @return true, if both are equal.
+	 */
+	public boolean isEqualItem( @Nullable final ItemStack left, @Nullable final ItemStack right )
+	{
+		return left != null && right != null && left.isItemEqual( right );
+	}
+
+	/**
+	 * Compares two {@link ItemStack} and their NBT tag for equality.
+	 * 
+	 * Use this when a precise check is required and the same item is required.
+	 * Not just something with different NBT tags.
+	 * 
+	 * @return true, if both are identical.
+	 */
+	public boolean isSameItem( @Nullable final ItemStack is, @Nullable final ItemStack filter )
+	{
+		return isEqualItem( is, filter ) && hasSameNbtTag( is, filter );
+	}
+
+	/**
+	 * Similar to {@link ItemComparisonHelper#isEqualItem(ItemStack, ItemStack)},
+	 * but it can further check, if both match the same {@link FuzzyMode}
+	 * or are considered equal by the {@link OreDictionary}
+	 * 
+	 * @param mode how to compare the two {@link ItemStack}s
+	 * @return true, if both are matching the mode or considered equal by the {@link OreDictionary}
+	 */
+	public boolean isFuzzyEqualItem( final ItemStack a, final ItemStack b, final FuzzyMode mode )
+	{
+		if( a == null && b == null )
+		{
+			return true;
+		}
+
+		if( a == null || b == null )
+		{
+			return false;
+		}
+
+		/*
+		 * if ( a.itemID != 0 && b.itemID != 0 && a.isItemStackDamageable() && ! a.getHasSubtypes() && a.itemID ==
+		 * b.itemID ) { return (a.getItemDamage() > 0) == (b.getItemDamage() > 0); }
+		 */
+
+		// test damageable items..
+		if( a.getItem() != null && b.getItem() != null && a.getItem().isDamageable() && a.getItem() == b.getItem() )
+		{
+			try
+			{
+				if( mode == FuzzyMode.IGNORE_ALL )
+				{
+					return true;
+				}
+				else if( mode == FuzzyMode.PERCENT_99 )
+				{
+					final Item ai = a.getItem();
+					final Item bi = b.getItem();
+
+					return ( ai.getDurabilityForDisplay( a ) > 1 ) == ( bi.getDurabilityForDisplay( b ) > 1 );
+				}
+				else
+				{
+					final Item ai = a.getItem();
+					final Item bi = b.getItem();
+
+					final float percentDamagedOfA = 1.0f - (float) ai.getDurabilityForDisplay( a );
+					final float percentDamagedOfB = 1.0f - (float) bi.getDurabilityForDisplay( b );
+
+					return ( percentDamagedOfA > mode.breakPoint ) == ( percentDamagedOfB > mode.breakPoint );
+				}
+			}
+			catch( final Throwable e )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1181)